ui/src/features/components/Report/compareReports.js

```diff
@@ -72,6 +72,44 @@ class CompareReports extends React.Component {
         this.setMergedReports(reportsList);
     };
 
+    onEditSymbol = (value, index) => {
+        const {reportsList} = this.state;
+        const {aggregateReports} = this.props;
+        const newAggregateReports = aggregateReports;
+        const currentReport = newAggregateReports[index];
+        const originalAlias = currentReport.alias;
+        currentReport.alias = value;
```

Comment:
it's a bad practice to change the data of prop, and we use redux as "a single source of truth", so I propose to change the name as a part of the selector (by saving the names also in redux) or, the more easy way is to convert the aggregateReports to state with the new names, so the component will read the data with the new names from the state - you can use getDerivedStateFromProps() lifecycle

Reply:
@manorlh wdyt? can we get this merged?